server/src/main/java/org/opensearch/common/cache/tier/EhCacheDiskCachingTier.java

```diff
@@ -40,13 +43,25 @@
 import org.ehcache.expiry.ExpiryPolicy;
 import org.ehcache.impl.config.store.disk.OffHeapDiskStoreConfiguration;
 
+/**
+ * This ehcache disk caching tier uses its value serializer outside ehcache.
+ * Values are transformed to byte[] outside ehcache and then ehcache uses its bundled byte[] serializer.
+ * The key serializer you pass to this class produces a byte[]. This serializer is passed to a wrapper which
+ * implements Ehcache's serializer implementation and produces a BytesBuffer. The wrapper instance is then passed to ehcache.
+ * This is done because to get keys on a disk tier, ehcache internally checks the equals() method of the serializer,
+ * but ALSO requires newKey.equals(storedKey) (this isn't documented), which is the case for ByteBuffer but not byte[].
+ * This limitation means that the key serializer must preserve the class of the key before/after serialization,
+ * but the value serializer does not have to do this.
+ * @param <K> The key type of cache entries
+ * @param <V> The value type of cache entries
+ */
 public class EhCacheDiskCachingTier<K, V> implements DiskCachingTier<K, V> {
 
     // A Cache manager can create many caches.
     private final PersistentCacheManager cacheManager;
 
     // Disk cache
-    private Cache<K, V> cache;
+    private Cache<K, byte[]> cache;
```

Comment:
This looks confusing. Where we take key as generic K but value as byte[]. Ideally we need to serialize both eventually. Plus it can happen that both key and value maybe of same type like ByteReference, String etc, so having this hybrid doesn't make sense.

Better to keep both byte[] or use generic for value as well and define a valueWrapper like keyWrapper.

Let me know it it make sense.